src/logger.ts

```diff
@@ -88,22 +91,35 @@ export class Logger {
     for (let i = 0; i < options.levels.length; i++) {
       const level = options.levels[i];
       if (i <= levelIndex) {
-        this[level] = (...args) => {
+        // ts: this doesn't have an index signature, but we want to set
+        // properties anyway.
+        // tslint:disable-next-line:no-any
+        (this as any)[level] = (...args: any[]) => {
```

Comment:
Are `options.levels` limited to info, warn, error, debug, silent, and silly? Or is any string allowed? If the latter, I don't see the point of separating out `CustomLevelsLogger`.

Reply:
Separating out `CustomLevelsLogger` makes it impossible to mock or extend the logger in a useful way.

Using custom log levels is a use case that is not useful for any Google-authored module anyway, because they always use the defaults.